src/renderers/shared/fiber/ReactChildFiber.js

```diff
@@ -200,6 +200,16 @@ function throwOnInvalidObjectType(returnFiber: Fiber, newChild: Object) {
   }
 }
 
+function warnOnFunctionType() {
+  warning(
+    false,
```

Comment:
To avoid two extra functions calls in DEV. (It's small but can add up.)

Also I find our `warning` notation with falsy condition meaning a failure very confusing. I try to avoid it whenever I write warnings, and always use `warning(false` so it’s clear from outer condition in which case it executes.